src/index.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import injectSheetDefault from 'react-jss'
+
+
+type StyledElementType = Function & { Tag: string, styles: Object }
+type PrimitiveProps = {
+  classes: Object,
+  children?: any,
+  className?: string,
+}
+
+
+export const prepareStyled = (injectSheet?: Function = injectSheetDefault) =>
+  (
+    Element: string | StyledElementType,
+    styles: Object,
+    baseStyles?: Object = {},
+  ): StyledElementType => {
+    const {
+      Tag,
+      styles: inheritStyles = {},
+    }: {
+      Tag: string,
+      styles?: Object
+    } = typeof Element === 'string' ? { Tag: Element } : Element
+
+    const elementStyles = { ...inheritStyles, ...styles }
```

Comment:
In this case we increase amount of styles rendered to css, we should think how to leverage `jss-compose` for this.